### PR TITLE
Remove SyncedCollections code (bug 747576)

### PR DESCRIPTION
**Pull request by diox**
The collector Add-on which is supposed to be used with that is no
longer maintained: https://blog.mozilla.org/addons/2012/03/07/add-on-collector-retiremen/

### CODE REVIEW

File: apps/discovery/views.py [hunk 1]
```diff
@@ -194,43 +191,10 @@ def recommendations(request, version, platform, limit=9, compat_mode=None):
...
     index = Collection.make_index(addon_ids)
-
     ids, recs = Collection.get_recs_from_ids(addon_ids, request.APP, version,
                                              compat_mode)
     recs = _recommendations(request, version, platform, limit, index, ids,
                             recs, compat_mode)
-
-    # We're only storing a percentage of the collections we see because the db
-    # can't keep up with 100%.
-    if not waffle.sample_is_active('disco-pane-store-collections'):
```
Comment (magopian) [inline apps/discovery/views.py]: Do we want a migration to remove that waffle too? Or is it used elsewhere?

**Comment by magopian** [comment]
r+wc, thanks!

**Comment by EnTeQuAk** [comment]
Looks good :cake:

File: apps/discovery/views.py (hunk 1 above)
Comment (diox) [inline apps/discovery/views.py]: Good call, I'll add that

**Comment by magopian** [comment]
Need to update the migration number, some migrations have landed on master since then.

**Comment by diox** [comment]
Migration amended to add waffle removal, m2m table and updated to use new migration number 847.

**Pull request merged**